### PR TITLE
Issue #4592: Added AbstractPathTestSupport, AbstractModuleTestSupport, AbstractIndentationTestSupport and all tests in ITs extend them

### PR DESCRIPTION
**Pull request by subkrish**
Issue #4592 

This PR creates `AbstractPathTestSupport`, `AbstractModuleTestSupport`, `AbstractIndentationTestSupport`. All tests in ITs now extend these test supports.

### CODE REVIEW

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java [hunk 1]
```diff
@@ -133,33 +133,20 @@ protected DefaultConfiguration createCheckerConfig(Configuration config) {
...
+     * method inside.
      * @param config configuration.
      * @param fileName file name to verify.
      * @param expected an array of expected messages.
      * @param warnsExpected an array of expected warning numbers.
      * @throws Exception if exception occurs during verification process.
      */
     protected void verify(Configuration config, String fileName, String[] expected,
-            Integer... warnsExpected) throws Exception {
+                          Integer... warnsExpected) throws Exception {
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: Please undo superfluous change.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java [hunk 2]
```diff
@@ -175,10 +162,10 @@ protected void verify(Configuration config, String fileName, String[] expected,
...
      */
     protected void verify(Checker checker,
-            File[] processedFiles,
-            String messageFileName,
-            String[] expected,
-            Integer... warnsExpected)
+                          File[] processedFiles,
+                          String messageFileName,
+                          String[] expected,
+                          Integer... warnsExpected)
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: same.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java [hunk 3]
```diff
@@ -205,7 +192,7 @@ protected void verify(Checker checker,
                 final int lineNumber = Integer.parseInt(parseInt);
                 assertTrue("input file is expected to have a warning comment on line number "
                         + lineNumber, previousLineNumber == lineNumber
-                            || theWarnings.remove((Integer) lineNumber));
+                        || theWarnings.remove((Integer) lineNumber));
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: same.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java [hunk 4]
```diff
@@ -225,7 +212,7 @@ protected void verify(Checker checker,
      * @param arguments  the arguments of message in 'messages.properties' file.
      */
     protected String getCheckMessage(Class<? extends AbstractViolationReporter> aClass,
-            String messageKey, Object... arguments) {
+                                     String messageKey, Object... arguments) {
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: same.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java [hunk 5]
```diff
@@ -246,7 +233,7 @@ protected String getCheckMessage(Class<? extends AbstractViolationReporter> aCla
      * @param arguments the arguments of message in 'messages.properties' file.
      */
     protected String getCheckMessage(Map<String, String> messages, String messageKey,
-            Object... arguments) {
+                                     Object... arguments) {
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: same.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java [hunk 6]
```diff
@@ -300,14 +287,14 @@ protected static Configuration getCheckConfig(String checkName, String checkId)
                     throw new IllegalStateException("problem to get ID attribute from " + conf, ex);
                 }
             })
-            .findFirst().orElseGet(null);
+                    .findFirst().orElseGet(null);
```
Comment (rnveach) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: same.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java (hunk 1 above)
Comment (subkrish) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: @rnveach Done.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java (hunk 2 above)
Comment (subkrish) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: @rnveach Done.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java (hunk 3 above)
Comment (subkrish) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: @rnveach Done.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java (hunk 4 above)
Comment (subkrish) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: @rnveach Done.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java (hunk 5 above)
Comment (subkrish) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: @rnveach Done.

File: src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java (hunk 6 above)
Comment (subkrish) [inline src/it/java/com/google/checkstyle/test/base/AbstractModuleTestSupport.java]: @rnveach Done.

**Comment by rnveach** [comment]
@romani Once this PR is merged, we still have to make modifications to the IT abstract classes so that it supports Checks as well as FileSets, similar what I did for tests.
Example: https://github.com/checkstyle/checkstyle/blob/master/src/it/java/com/google/checkstyle/test/chapter2filebasic/rule231filetab/FileTabCharacterTest.java#L41
We shouldn't need special code like this.

**Comment by subkrish** [comment]
@rnveach @romani The issue is basically complete. But just the 
> we still have to make modifications to the IT abstract classes so that it supports Checks as well as FileSets, similar what I did for tests.

is remaining. With that the issue is complete and can be closed that is done.

**Pull request merged**